src/engine.rs

```diff
@@ -441,7 +456,16 @@ impl Reedline {
         loop {
             let mut paste_enter_state = false;
 
-            if event::poll(Duration::from_millis(1000))? {
+            #[cfg(feature = "external_printer")]
+            if let Some(ref external_printer) = self.external_printer {
+                if let Ok(line) = external_printer.receiver().try_recv() {
+                    self.painter.print_crlf()?;
```

Comment:
I don't know if there might be a good usecase for this printer to not automatically act as `println` compared to `print` but yeah implementing the latter correctly is probably unevenly harder.

What I am unsure about. When invoking this we don't have a nicely defined cursor position (We use the regular ANSI terminal cursor as the user cursor.) If the external content get's printed after the cursor, nothing is stopping the editor from overwriting that output.

Reply:
Here is my small demo:
https://github.com/GrxE/reedline_with_external_printer